src/relic.cpp

```diff
@@ -261,22 +262,38 @@ void relic_charge_info::serialize( JsonOut &jsout ) const
     jsout.member( "charges", charges );
     jsout.member( "charges_per_use", charges_per_use );
     jsout.member( "max_charges", max_charges );
+    jsout.member( "regenerate_ammo", regenerate_ammo );
     jsout.member( "recharge_type", type );
     jsout.member( "activation_accumulator", activation_accumulator );
     jsout.member( "time", activation_time );
     jsout.end_object();
 }
 
-void relic_charge_info::accumulate_charge()
+void relic_charge_info::accumulate_charge( item &parent )
 {
-    if( charges >= max_charges || activation_time == 0_seconds ) {
+    if( activation_time == 0_seconds ) {
         // return early, no accumulation required
         return;
+    } else if( regenerate_ammo ) {
+        if( parent.remaining_ammo_capacity() == 0 ) {
+            return;
+        }
```

Comment:
```suggestion
    } else if( regenerate_ammo && parent.remaining_ammo_capacity() == 0 ) {
        return;
    }
```

Reply:
i had a lot of issues with that so i had to expand it out so i could reason it out. i'll experiment a little bit more with simplification a little bit later.